department-of-veterans-affairs/abd-vro
# MCP-1988: broadcast audit messages to DB and slack

## What was the problem?

We need the ability to route audit events to multiple recipients. One of them is a slack channel to report exceptions.

Associated tickets or Slack threads:

- [MCP-1988](https://amida.atlassian.net/browse/MCP-1988)

## How does this fix it?[^1]
Restructured event infrastructure so that:
- events are automatically on exception
- every route is captured automatically if the route id starts with "mas"
- audit events can have multiple recipient lists

## How to test this PR

Testing this is a bit of challenge. Here are the steps that can be performed in a local dev env.

- Step 1: Set these properties to a valid slack webhook and channel

slack:
  exception:
    webhook: "${SLACK_EXCEPTION_WEBHOOK}"
    channel: "${SLACK_EXCEPTION_CHANNEL}"

- Step 2: Cause the application to throw an exception during claim processing. I don't know how to do this "naturally". The code can be temporarily modified to throw an exception

If steps 1 and 2 are done correctly, you will get a notification on the slack channel with the exception and the route. where it occured


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
